add notebook about creating netcdf with Fortran an Python

ScottWales opened the pull request.

Commit:
add notebook about creating netcdf with Fortran an Python

ScottWales commented on notebooks/2018-10-05-create-netcdf.ipynb:
````diff
@@ -0,0 +1,368 @@
+    "    status = nf90_put_att(ncid, varid_field, '_FillValue', -2e8)\n",
+    "```\n",
+    "\n",
+    "We can add any number of attributes to each variable, or even to the global file. They can be of any common type. For a list of attribute conventions, you can look here: https://www.unidata.ucar.edu/software/netcdf/docs/attribute_conventions.html\n",
````
Instead point to relevant section of http://cfconventions.org/

HoWol76 replied:
Good point

ScottWales commented on notebooks/2018-10-05-create-netcdf.ipynb:
```diff
@@ -0,0 +1,368 @@
+    "\n",
+    "### create an empty file\n",
+    "\n",
+    "To create a file, we need a file handle, an integer variable, by convention `ncid`, which, similar to the `unit` of `read` and `write` commands, references the file we created:\n",
```
May be worth pointing out opening an existing file works differently

HoWol76 replied:
I added a note to the beginning of the document.

ScottWales commented on notebooks/2018-10-05-create-netcdf.ipynb:
````diff
@@ -0,0 +1,368 @@
+    "Since we have now finished with the header, we can tell the netcdf API about that:\n",
+    "\n",
+    "```\n",
+    "    status = nf90_enddef(ncid)\n",
````
Only needed for netcdf3 / classic mode files

HoWol76 replied:
Again, my documentation does not make this distinction. (10 year old documentation, though.) Must test.

ScottWales commented on notebooks/2018-10-05-create-netcdf.ipynb:
```diff
@@ -0,0 +1,368 @@
+    "    implicit none\n",
+    "    integer :: status, ncid\n",
+    "\n",
+    "    status = nf90_create('data.nc', NF90_HDF5, ncid)\n",
```
`NF90_HDF5` is not a documented creation flag - should be `NF90_NETCDF4`? https://www.unidata.ucar.edu/software/netcdf/docs/group__datasets.html#ga427f5a0b24f1d426a99bcc37b8a39cac

HoWol76 replied:
This is interesting. I have only found this Fortran API documentation: https://www.unidata.ucar.edu/software/netcdf/netcdf-4/newdocs/netcdf-f90/index.html#Top -- it is 10 years old, and it does not contain `NF90_NETCDF4`. The link you point to shows a C routine, not a Fortran routine.

ScottWales replied:
fortran routines are equivalent to c, with a changed prefix

HoWol76 replied:
Fixed in latest version

ScottWales commented on notebooks/2018-10-05-create-netcdf.ipynb:
```diff
@@ -0,0 +1,368 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "field.latitude.attrs = {'units':'degree_north'}\n",
```
This overwrites existing attributes, prefer `field.latitude.attrs['units'] = 'degrees_north'`

HoWol76 replied:
Interesting, I thought I had tried that and got an error message, but now it works. Maybe I had `.attr['units']` or some typo like that. I'll do that.

Commit:
changed date from 5/10/18 to 12/10/18

Commit:
incorporated some of Scott's comments

Commit:
incorporate advice from Scott and Claire

Scott recommended amongst other things to use the C API documentation
instead of the outdated F90 API documentation

Claire asked to include compression in the discussion.

Commit:
Merge branch 'master' into holger

Commit:
typo and revamp of enddef

Calls to `nf90_enddef` are no longer needed with NetCDF4.
The text now reflects that.

Commit:
Text modifications

Commit:
Bug: mix up between nx and ny in creating the coordinates.

ccarouge commented:
I've put a couple of commits in holger_review branch.
Most are text modifications (first commit). Second commit is a bug with a mix up between nx and ny.

ccarouge commented:
One more thing. You do `ncdump -hs` for the file created by Python and `ncdump -h` for the file created by Fortran. If people aren't very careful, they might think netcdf files created by Python contain more info than if created by Fortran. So please choose one or the other.

Commit:
incorporate edits from Claire

Some typos, also using the same ncdump for both created files to avoid
confusion.

Commit:
tweak to make the two netcdf files similar

Before, they had different chunking, and the order
of the dimensions was reversed.
